PMMain/Midas/MidasGUIQTest/KeyboardSettingFileReader_Test.cpp

```diff
@@ -1,12 +1,84 @@
 #include "KeyboardSettingsReader.h"
 #include "RingData.h"
 #include <gtest\gtest.h>
+#include <iostream>
+#include <fstream>
+
+TEST(KeyboardFileReaderTest1, testSize) {
+
+    std::ofstream outputFile("testFile.txt");
+
+    if (!outputFile)
+    {
+        //error in creating the file
+        perror("Error creating file");
+    }
+    //populating test file
+    outputFile << "aA bB cC dD eE fF gG hH iI jJ kK lL mM" << std::endl;
+    outputFile << "nN oO pP qQ rR sS tT uU vV wW xX yY zZ" << std::endl;
+    outputFile << "`~ 1! 2@ 3# 4$ 5% 6^ 7& 8* 9( 0) -_ =+" << std::endl;
+    outputFile << "[{ ]} ;: ' \ | , < .> | \ / ?" << std::endl;
+    outputFile.close();
+
+    std::vector<ringData>  test;
+    KeyboardSettingsReader reader;
+    reader.readKeyboardSetupFile(test, "testFile.txt");
+
+
+    if (remove("testFile.txt") != 0)
+    {
+        perror("Error deleting file");
+    }
+    else
+    {
+        puts("File successfully deleted");
+    }
+
+
+    ASSERT_EQ(test.size(), 2); //checks for # of rings
+
+}
+
+TEST(KeyboardFileReaderTest1, testValue) {
+
+
+    std::ofstream outputFile("testFile.txt");
+
+    if (!outputFile)
+    {
+        //error in creating the file
+        perror("Error creating file");
+    }
+    //populating test file
+    outputFile << "aA bB cC dD eE fF gG hH iI jJ kK lL mM" << std::endl;
+    outputFile << "nN oO pP qQ rR sS tT uU vV wW xX yY zZ" << std::endl;
+    outputFile << "`~ 1! 2@ 3# 4$ 5% 6^ 7& 8* 9( 0) -_ =+" << std::endl;
+    outputFile << "[{ ]} ;: ' \ | , < .> | \ / ?" << std::endl;
+    outputFile.close();
```

Comment:
If this same file is being created for multiple tests, it may be worth moving it to a helper function that creates the test file.